pkg/libvmi/vmi.go

```diff
@@ -84,6 +84,42 @@ func WithNamespace(namespace string) Option {
 	}
 }
 
+func WithUserPasswordAccessCredential(secretName string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		vmi.Spec.AccessCredentials = []v1.AccessCredential{{
+			UserPassword: &v1.UserPasswordAccessCredential{
+				Source: v1.UserPasswordAccessCredentialSource{
+					Secret: &v1.AccessCredentialSecretSource{
+						SecretName: secretName,
+					},
+				},
+				PropagationMethod: v1.UserPasswordAccessCredentialPropagationMethod{
+					QemuGuestAgent: &v1.QemuGuestAgentUserPasswordAccessCredentialPropagation{},
+				},
+			},
+		}}
+	}
+}
```

Comment:
We should go in append, so that the option can be used several times:
```suggestion
func WithSSHPublicKeyAccessCredential(secretName string, users []string) Option {
	return func(vmi *v1.VirtualMachineInstance) {
		vmi.Spec.AccessCredentials = append(vmi.Spec.AccessCredentials, v1.AccessCredential{
			SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
				Source: v1.SSHPublicKeyAccessCredentialSource{
					Secret: &v1.AccessCredentialSecretSource{
						SecretName: secretName,
					},
				},
				PropagationMethod: v1.SSHPublicKeyAccessCredentialPropagationMethod{
					QemuGuestAgent: &v1.QemuGuestAgentSSHPublicKeyAccessCredentialPropagation{
						Users: users,
					},
				},
			},
		})
	}
}
```